lib/variant-model.js

```diff
@@ -45,17 +47,42 @@ _.extend(Variant.prototype, {
     return this;
   },
 
-  respondWithFile: function(fileName) {
+  respondWithFile: function(options) {
+    options = _.extend({
+      variant_id: this.id(),
+      path: this._route.path(),
+    }, options);
+
     return this.respondWith(function(request, reply) {
-      var self = this;
-      fileName = fileName.replace(/\{([^\}]+)\}/g, function(match, token) {
-        var val = request.params[token];
-        if (!val) {
-          val = self.state(token);
+      var smocks = require('./index');
+
+      options = _.extend({
+        fileName: smocks.initOptions.fileName || 'response.json',
+        mockDir: smocks.initOptions.mockDir || './mocked-data',
+        statusCode: smocks.initOptions.statusCode || 200
+      }, options);
+
+      var pattern = /{\w+}\/*/g;
+      var path = options.path.replace(pattern, '');
+      var dataVariantFilePath = Path.join(options.mockDir, path, options.variant_id, options.fileName);
+
+      // Search variant directory
+      Fs.stat(dataVariantFilePath, function (err) {
```

Comment:
just open the file and 404 on error.  write the stream buffer straight out.
```
fs.readFile(filePath, function (err, data) {
  if (err) {
    reply().code(404);
  }
  reply(data).type('application/json').code(options.statusCode);
});
```

rinse wash repeat for the multiple optional paths described in the earlier comment